### ISSUE TITLE
Conflict with prettier format with brackets

### ISSUE THREAD

**Issue by abcfy2** [ansible/ansible-lint#2678]
##### Summary

Here is the sample (After prettier format):

```yml
- name: "Configure Nginx"
  ansible.builtin.copy:
    src: "{{ item.src }}"
    dest: "{{ item.dest }}"
    owner: root
    group: root
    mode: "0644"
  notify:
    - "Restart Nginx"
  with_items:
    - {
        src: systemd_nginx_override.conf,
        dest: /etc/systemd/system/nginx.service.d/override.conf,
      }
    - { src: nginx.conf, dest: /etc/nginx/ }
    - { src: brotli.conf, dest: /etc/nginx/conf.d/ }
```

Will get this error:

```sh
yaml[indentation]: Wrong indentation: expected 6 but found 8
tasks/main.yml:121

yaml[indentation]: Wrong indentation: expected 6 but found 8
tasks/main.yml:122

yaml[indentation]: Wrong indentation: expected 4 but found 6
tasks/main.yml:123
```

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible [core 2.14.0]
  config file = /home/fengyu/projects/devops/ansible_playbooks/ansible.cfg
  configured module search path = ['/home/fengyu/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.10/site-packages/ansible
  ansible collection location = /home/fengyu/.ansible/collections:/usr/share/ansible/collections
  executable location = /usr/bin/ansible
  python version = 3.10.8 (main, Nov  1 2022, 14:18:21) [GCC 12.2.0] (/usr/bin/python)
  jinja version = 3.1.2
  libyaml = True

ansible-lint --version
ansible-lint 6.8.6 using ansible 2.14.0
```

- ansible installation method: one of source, pip, OS package
- ansible-lint installation method: one of source, pip, OS package

##### STEPS TO REPRODUCE

```yml
- name: "Configure Nginx"
  ansible.builtin.copy:
    src: "{{ item.src }}"
    dest: "{{ item.dest }}"
    owner: root
    group: root
    mode: "0644"
  notify:
    - "Restart Nginx"
  with_items:
    - {
        src: systemd_nginx_override.conf,
        dest: /etc/systemd/system/nginx.service.d/override.conf,
      }
    - { src: nginx.conf, dest: /etc/nginx/ }
    - { src: brotli.conf, dest: /etc/nginx/conf.d/ }
```

```console (paste below)
yaml[indentation]: Wrong indentation: expected 6 but found 8
tasks/main.yml:121

yaml[indentation]: Wrong indentation: expected 6 but found 8
tasks/main.yml:122

yaml[indentation]: Wrong indentation: expected 4 but found 6
tasks/main.yml:123
```

**Comment by ssbarnea** [comment]
You failed to mention the input data before feeding it to prettier. If that was passing out linting it would be a valid but, if not, is not our problem, see [1](https://en.wikipedia.org/wiki/Garbage_in,_garbage_out).

Use of inline dictionaries/mappings is discouraged in YAML anyway.

**Comment by abcfy2** [comment]
Thanks. Use YAML style dict will solve this issue:

```yaml
  with_items:
    - src: systemd_nginx_override.conf
      dest: /etc/systemd/system/nginx.service.d/override.conf
    - src: nginx.conf
      dest: /etc/nginx/
    - src: brotli.conf
      dest: /etc/nginx/conf.d/
```

**Comment by ssbarnea** [comment]
Yep, and that is much easier to read too!
